### PR TITLE
DynamoDB SessionHandler - Configurable data & session lifetime attr

### PR DESCRIPTION
**Pull request by alessionobile**
We needed to configure the DynamoDB Session data and session lifetime attributes as we share the session across different micro-services built upon different languages and frameworks and saving session data in different attributes than the ones supported by this default. Hence the PR:

- Added SessionConnectionConfigTrait with standard configurations defined both for Standard and Locking Session connections.
- Added tests for Standard and Custom configurations

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by alessionobile** [comment]
Hi @awstools @kstich

May I ask you some feedback on this PR? :)

Have another PR in a laravel-dynamodb-session driver that I'd like to submit once this one is approved.
https://github.com/alessionobile/laravel-dynamodb-session-driver/commit/a53c9855d4b25b9940f6e4fd2ad3e2dc555d05bf

Thanks for your great work guys

File: src/DynamoDb/SessionHandler.php [hunk 1]
```diff
@@ -135,14 +140,17 @@ public function read($id)
         // PHP expects an empty string to be returned from this method if no
         // data is retrieved
         $this->dataRead = '';
-
+
```
Comment (srchase) [inline src/DynamoDb/SessionHandler.php]: nit: spacing

File: src/DynamoDb/SessionHandler.php [hunk 2]
```diff
@@ -44,13 +44,18 @@ class SessionHandler implements \SessionHandlerInterface
      * The configuration array accepts the following array keys and values:
      * - table_name:               Name of table to store the sessions.
      * - hash_key:                 Name of hash key in table. Default: "id".
+     * - data_attribute:           Name of the data attribute in table. Default: "data".
      * - session_lifetime:         Lifetime of inactive sessions expiration.
+     * - session_lifetime_attribute: Name of the session life time attribute in table. Default: "expires".
```
Comment (srchase) [inline src/DynamoDb/SessionHandler.php]: nit: spacing

File: src/DynamoDb/SessionConnectionConfigTrait.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
+<?php
+namespace Aws\DynamoDb;
+
+trait SessionConnectionConfigTrait
+{
+    /** @var string Name of table to store the sessions */
+    protected $table_name = 'sessions';
```
Comment (srchase) [inline src/DynamoDb/SessionConnectionConfigTrait.php]: All of these variables should be camelCased.

**Comment by alessionobile** [comment]
Thanks for the feedback @srchase 
Spacing fixed

File: src/DynamoDb/SessionConnectionConfigTrait.php (hunk 3 above)
Comment (alessionobile) [reply src/DynamoDb/SessionConnectionConfigTrait.php]: @srchase regarding the tests - totally agree with you.
I actually covered both standard and custom tests here:
https://github.com/alessionobile/aws-sdk-php/blob/master/tests/DynamoDb/StandardSessionConnectionTest.php#L18
and here:
https://github.com/alessionobile/aws-sdk-php/blob/master/tests/DynamoDb/StandardSessionConnectionTest.php#L34

But perhaps if you think is best i can create a specific test case for it

Comment (alessionobile) [reply src/DynamoDb/SessionConnectionConfigTrait.php]: @srchase added a specific test for the ConfigTrait.
https://github.com/alessionobile/aws-sdk-php/blob/master/tests/DynamoDb/SessionConnectionConfigTraitTest.php

I kept the standard/custom config test anyway within the StandardConnection test case. What do you think? :) Better to keep it there or take it away and rely only in SessionConnectionConfigTraitTest.php?

**Comment by alessionobile** [comment]
hi @srchase is there a way to re-run codecov? :)

File: src/DynamoDb/SessionHandler.php [hunk 4]
```diff
@@ -139,10 +144,13 @@ public function read($id)
         // Get session data using the selected locking strategy
         $item = $this->connection->read($this->formatId($id));
 
+        $data_attribute = $this->connection->getDataAttribute();
```
[the remark this answers is not in the document]
Comment (alessionobile) [reply src/DynamoDb/SessionHandler.php]: Absolutely :) Sorry missed them out. Just pushed a commit

**Comment by alessionobile** [comment]
hi @srchase, hope you are doing well. 
Is there anything else needed? :)

**Comment by srchase** [comment]
Hello @alessionobile!

Could you rebase your PR so Codecov will properly reflect the test coverage?

**Comment by alessionobile** [comment]
@srchase absolutely :) i created a new pr https://github.com/aws/aws-sdk-php/pull/1784

With full rebase and squashed everything in 1 commit :)

**Pull request closed without merging**